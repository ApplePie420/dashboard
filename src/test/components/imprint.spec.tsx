import { fireEvent, render } from "@testing-library/react";
import Imprint, { IImprintProps, ImprintField } from "../../components/imprint";

const props: IImprintProps = {
  name: {
    text: "Test Name",
    link: "#",
  },
  address: {
    text: "Test Address",
    link: "#",
  },
  phone: {
    text: "Test Phone",
    link: "#",
  },
  email: {
    text: "Test Email",
    link: "#",
  },
  url: {
    text: "Test URL",
    link: "#",
  },
  text: "This is a test!",
};

describe("imprint.tsx", () => {
  const location: Location = window.location;

  beforeEach(() => {
    // @ts-ignore
    delete window.location;

    window.location = {
      ...location,
      reload: jest.fn(),
    };
  });

  it("Tests Imprint", () => {
    const { asFragment } = render(<Imprint imprint={props} />);
    expect(asFragment).toMatchSnapshot();
  });

  it("Tests #imprint", () => {
    const location = window.location.href;
    window.location.href = location + "#imprint";

    const imprintModal = render(<Imprint imprint={props} />);

    fireEvent.click(imprintModal.getByTestId("toggle-button"));
    //fireEvent.click(imprintModal.getByTestId("close-button"));

    expect(window.location.href).toEqual(location);
  });

  it("Tests ImprintField", () => {
    const { asFragment } = render(<ImprintField field={props.name} />);
    expect(asFragment).toMatchSnapshot();
  });
});
