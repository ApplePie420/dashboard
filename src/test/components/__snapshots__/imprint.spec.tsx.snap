// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`imprint.tsx Tests Imprint 1`] = `
<body>
  <div>
    <div
      class="sc-bczRLJ kimCm"
    >
      <h2
        class="sc-gsnTZi jiVNCT"
      >
        About
      </h2>
      <ul
        class="sc-hKMtZM jEBHIA"
      >
        <div
          class="sc-evZas dWbjhp"
        >
          <h3
            class="sc-dkzDqf dzkwjK"
          >
            Imprint
          </h3>
          <div
            class="sc-ftvSup cpmMIl"
            hidden=""
          >
            <div
              class="sc-jqUVSM izDwOx"
            >
              <h2
                class="sc-gsnTZi jiVNCT"
              >
                Legal Disclosure
              </h2>
              <button
                class="sc-gKXOVf inaviU"
                data-testid="close-button"
              >
                <i
                  class="sc-iBkjds fflOgg"
                >
                  close
                </i>
              </button>
            </div>
            <div>
              <h3
                class="sc-dkzDqf sc-kDDrLX dzkwjK jYBHXB"
              >
                Information in accordance with section 5 TMG
              </h3>
              <a
                class="sc-crXcEl hLNarr"
                href="#"
              >
                Test Name
              </a>
              <a
                class="sc-crXcEl hLNarr"
                href="#"
              >
                Test Address
              </a>
              <a
                class="sc-crXcEl hLNarr"
                href="#"
              >
                Test Email
              </a>
              <a
                class="sc-crXcEl hLNarr"
                href="#"
              >
                Test Phone
              </a>
              <a
                class="sc-crXcEl hLNarr"
                href="#"
              >
                Test URL
              </a>
            </div>
            <div>
              <h3
                class="sc-dkzDqf sc-kDDrLX dzkwjK jYBHXB"
              >
                Imprint
              </h3>
              <p
                class="sc-iqcoie fwpsgH"
              >
                This is a test!
              </p>
            </div>
          </div>
          <p
            class="sc-papXJ gkwCsw"
            data-testid="toggle-button"
          >
            View Imprint
          </p>
        </div>
      </ul>
    </div>
  </div>
</body>
`;

exports[`imprint.tsx Tests ImprintField 1`] = `
<body>
  <div>
    <a
      class="sc-crXcEl hLNarr"
      href="#"
    >
      Test Name
    </a>
  </div>
</body>
`;
